processor/src/main/java/org/mapstruct/ap/internal/processor/creation/MappingResolverImpl.java

```diff
@@ -640,15 +433,21 @@ private boolean isUpdateMethodForMapping(Method methodCandidate) {
                 && !methodCandidate.isLifecycleCallbackMethod();
         }
 
-        private <T extends Method> SelectedMethod<T> getBestMatch(List<T> methods, Type sourceType, Type returnType) {
+        private <T extends Method> List<SelectedMethod<T>> getBestMatch(Set<T> methods, Type source, Type target) {
```

Comment:
Why do we need 2 methods? One with `Set` and another one with `List`?

Reply:
I do use sets in the 2 step matchers to store unique 'SelectedMethods'.. I don't do that in the regular one-step variant. So I wanted to preserve order there (did not want to touch it). I bolted on the Set and did not want to change it all over the place.. 

Perhaps later in a cleanup?

Reply:
I'll push an update.. removing the method.. I'll change it at the (one) call site.